js/lib/cli.js

```diff
@@ -293,24 +293,26 @@ function makePretty(code, config, outfile, callback) {
         var fileType = getOutputType(outfile, config.type);
         var pretty = beautify[fileType](code, config);
 
-        callback(null, pretty, outfile, config);
+        callback(null, pretty, outfile, config, code);
     } catch (ex) {
         callback(ex);
     }
 }
 
-function writePretty(err, pretty, outfile, config) {
+function writePretty(err, pretty, outfile, config, code) {
     if (err) {
         console.error(err);
         process.exit(1);
     }
 
     if (outfile) {
-        try {
-            fs.writeFileSync(outfile, pretty, 'utf8');
-            logToStdout('beautified ' + path.relative(process.cwd(), outfile), config);
-        } catch (ex) {
-            onOutputError(ex);
+        if (code !== pretty) {
```

Comment:
Here's the problem.  You can't just not write the code if it is the same.  You need to check (around line 267) if the input and output path are the same.  If so, you can check here if the input and output code is unchanged and not change that file.  If the input path is empty, output path is empty (for piping cases), or if the input path differs from the output path you still have to write the output, even if it is unchanged.